apps/ex_wire/lib/ex_wire/kademlia.ex

```diff
@@ -1,85 +1,199 @@
 defmodule ExWire.Kademlia do
   @moduledoc """
-  Handles Kademlia algorithm state.
+  Handles Kademlia algorithm state. also a GenServer that manages Kademlia state.
```

Comment:
I think the separation of GenServer and domain logic is cleaner so I'm against these changes

Reply:
See https://pragdave.me/blog/2017/07/13/decoupling-interface-and-implementation-in-elixir.html

Reply:
I do not see any domain logic here, it's just spreading the process state over several modules. which is  what you would want to avoid (in my experience).

The other part that was also addressed was awkward server name passing in a Keyword type. Not used for anything else...

Reply:
and if you read closely to the link @ayrat555 this is not how they propose separating domain logic and gen servers. In their case, the beginning of the server would have had to been here: https://github.com/mana-ethereum/mana/blob/master/apps/ex_wire/lib/ex_wire/kademlia.ex
but it's not.

Reply:
Can't we split the difference and keep the server and logic separate, but now use a globally registered name (e.g. `@server_name __MODULE__`) to prevent having to thread the options through?

Reply:
something like #681?

Reply:
Yes, I think I like that a lot more :)